rules/windows/persistence_local_scheduled_task_scripting.toml

```diff
@@ -0,0 +1,42 @@
+[metadata]
+creation_date = "2020/11/29"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/11/29"
+
+[rule]
+author = ["Elastic"]
+description = "A scheduled task was created by a Windows script via cscript.exe, wscript.exe or powershell.exe, this can be used by an adversary to establish persistence."
+false_positives = ["Legitimate scheduled tasks may be created during installation of new software."]
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Scheduled Task Created by a Windows Script"
+note = "Decode the base 64 encoded Tasks Actions registry value to investigate the task configured action."
+risk_score = 43
+rule_id = "689b9d57-e4d5-4357-ad17-9c334609d79a"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Persistence"]
+type = "eql"
+
+query = '''
+sequence by host.id with maxspan = 5s
+  [library where file.name == "taskschd.dll" and process.name in ("cscript.exe", "wscript.exe", "powershell.exe")]
```

Comment:
Can we do any kind of detection hardening on the process names such as using original file names too? In case, malware or adversary copies/renames these files.

Reply:
not covered by default in imageload events, if we add to the sequence an extra block with process creation (where we require pe.ofn eq to xxx) it adds extra performance burden, we can cover renamed lolbins in a different rule where all commonly abused lolbins are grouped in one rule with min impact (only process events in the rule).